### PR TITLE
TypeScript init

### PR DESCRIPTION
**Pull request by DanielW1987**
The PR contains a number of preparations to be able to work with TypeScript in the future:

- Adjustment of the GitHub action on Pull Request: In addition, the frontend is now compiled TypeScript to JavaScript using different versions of Node.js to check for compatibility
- Configuration of webpack. The production build of the frontend is carried out by webpack. webpack bundles all TypeScript files of a page into a single minified JavaScript file. This happens when building the Docker Image, because the JavaScript files belong to the application as a static resource.

**Next Steps**
- Implement MyArtists page with TypeScript
- UnitTests for frontend

**Long-term roadmap**
- Completely decouple the frontend from the backend, e.g. by using Angular. The TypeScript files can be reused for the most part

### CODE REVIEW

File: webapp/src/main/resources/static/ts/src/blog/blog.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class BlogService {
+
+    constructor() {
+    }
+
+    printGreeting(): void {
+        console.log("Hello From Blog Service!");
```
Comment (hedgehog1833) [inline webapp/src/main/resources/static/ts/src/blog/blog.ts:7]: This should appear in the browser log?

Comment (DanielW1987) [reply webapp/src/main/resources/static/ts/src/blog/blog.ts]: You have to prepare your local environment to use the javascript that is compiled from typescript.

File: .github/workflows/compile-frontend.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  compile-frontend:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
```
[the remark this answers is not in the document]
Comment (DanielW1987) [reply .github/workflows/compile-frontend.yml:13]: to check for compatibility on pull request and only there.

**Comment by DanielW1987** [comment]
[TypeScript Proof of Concept](https://trello.com/c/2yXFnBfp/161-typescript-proof-of-concept)

**Comment by DanielW1987** [comment]
@hedgehog1833 You can now review.

**Pull request merged**